pkgs/development/libraries/qt-5/hooks/qmake-hook.sh

```diff
@@ -1,40 +1,36 @@
 . @fix_qmake_libtool@
 
-qmakeFlags=( ${qmakeFlags-} )
-
 qmakePrePhase() {
-    qmakeFlags_orig=( "${qmakeFlags[@]}" )
-
     # These flags must be added _before_ the flags specified in the derivation.
-    qmakeFlags=( \
-        "PREFIX=$out" \
-        "NIX_OUTPUT_OUT=$out" \
-        "NIX_OUTPUT_DEV=${!outputDev}" \
-        "NIX_OUTPUT_BIN=${!outputBin}" \
-        "NIX_OUTPUT_DOC=${!outputDev}/${qtDocPrefix:?}" \
-        "NIX_OUTPUT_QML=${!outputBin}/${qtQmlPrefix:?}" \
-        "NIX_OUTPUT_PLUGIN=${!outputBin}/${qtPluginPrefix:?}" \
-    )
+    prependToVar qmakeFlags \
+      "PREFIX=$out" \
+      "NIX_OUTPUT_OUT=$out" \
+      "NIX_OUTPUT_DEV=${!outputDev}" \
+      "NIX_OUTPUT_BIN=${!outputBin}" \
+      "NIX_OUTPUT_DOC=${!outputDev}/${qtDocPrefix:?}" \
+      "NIX_OUTPUT_QML=${!outputBin}/${qtQmlPrefix:?}" \
+      "NIX_OUTPUT_PLUGIN=${!outputBin}/${qtPluginPrefix:?}"
 
     if [ -n "@debug@" ]; then
-        qmakeFlags+=( "CONFIG+=debug" )
+        prependToVar qmakeFlags "CONFIG+=debug"
     else
-        qmakeFlags+=( "CONFIG+=release" )
+        prependToVar qmakeFlags "CONFIG+=release"
     fi
 
     # do the stripping ourselves (needed for separateDebugInfo)
-    qmakeFlags+=( "CONFIG+=nostrip" )
-
-    qmakeFlags+=( "${qmakeFlags_orig[@]}" )
+    prependToVar qmakeFlags "CONFIG+=nostrip"
```

Comment:
Because these are multiple `prependToVar` calls, I think you need to reverse the order of the prepend calls: `CONFIG+=nostrip` first, followed by `CONFIG+=debug` or release, then finally the other ones that are first in this diff.

Maybe it doesn't matter though.